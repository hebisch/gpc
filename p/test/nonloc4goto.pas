program test(output);

function trash(n: integer):integer;
begin
  if n>0 then
    trash := (2*trash(n-1)+1) mod 997
  else
    trash := 15
end;

procedure lvl1;
label 1;

begin
  begin
  var f :text;
  rewrite(f,'file.dat');
  writeln(f,'lvl1 called');
  begin
    writeln('lvl2 executing');
    goto 1;
  end;
  end;
1:
end;

begin
writeln('lvl0 called');
lvl1;
writeln('trash = ', trash(10000));
writeln('it worked');
end.
