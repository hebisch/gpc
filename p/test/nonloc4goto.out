lvl0 called
lvl2 executing
trash = 639
it worked
